lib/ui/settings/data/import/analyze_qr_code.dart

```diff
@@ -0,0 +1,268 @@
+  const QrScanner({super.key});
+
+  @override
+  State<QrScanner> createState() => _QrScannerState();
+}
+
+class _QrScannerState extends State<QrScanner> {
+  bool isNavigationPerformed = false;
+  bool isScannedByImage = false;
+
+  MobileScannerController scannerController = MobileScannerController(
+    detectionSpeed: DetectionSpeed.normal,
+    facing: CameraFacing.back,
+  );
+
+  @override
+  void dispose() {
+    scannerController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final l10n = context.l10n;
+    return SafeArea(
+      child: Scaffold(
+        backgroundColor: Colors.black,
+        body: Stack(
+          alignment: Alignment.topLeft,
+          children: [
+            Stack(
+              children: [
+                MobileScanner(
+                  controller: scannerController,
+                  onDetect: (capture) async {
+                    if (!isNavigationPerformed) {
+                      isNavigationPerformed = true;
+                      if (capture.barcodes[0].rawValue!
+                          .startsWith(kGoogleAuthExportPrefix)) {
+                        if (isScannedByImage) {
+                          final result = await showDialogWidget(
+                            context: context,
+                            title: l10n.reminderText,
+                            body: l10n.reminderPopupBody,
+                            buttons: [
+                              ButtonWidget(
+                                buttonType: ButtonType.primary,
+                                labelText: l10n.ok,
+                                isInAlert: true,
+                                buttonSize: ButtonSize.large,
+                                buttonAction: ButtonAction.first,
+                              ),
+                            ],
+                          );
+                          if (result?.action != null &&
+                              result!.action == ButtonAction.first) {
+                            isScannedByImage = false;
+                          }
+                        }
+                        HapticFeedback.vibrate();
+                        try {
+                          List<Code> codes =
+                              parseGoogleAuth(capture.barcodes[0].rawValue!);
+                          scannerController.dispose();
+                          Navigator.of(context).pop(codes);
+                        } catch (e) {
+                          showToast(context, l10n.parsingErrorText);
+                          Logger("Code parsing error").severe(
+                            "Error while parsing Google Auth QR code",
+                            e,
+                          );
+                          throw Exception(
+                            'Failed to parse Google Auth QR code \n ${e.toString()}',
+                          );
+                        }
```

Comment:
Can you please test and confirm what happens on the UI when exception is thrown here?

Reply:
Exception thing does not give anything. Toast is coming @ua741